### PR TITLE
S2t4 sign in view

### PR DESCRIPTION
**Pull request by bentleybiscuit**

### CODE REVIEW

File: src/Components/Instructions/index.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+
+export class Instructions extends React.Component {
```
Comment (sudokufan) [inline src/Components/Instructions/index.js]: I didn't know you could directly export a class like this. Every day's a learning day!

File: src/Components/LandingPage/index.css [hunk 2]
```diff
@@ -1,12 +1,70 @@
-.visitor {
-    position: fixed;
-    bottom: 45%;
-    right: 40%;
+header {
+    width: 100vw;
+
+}
```
Comment (sudokufan) [inline src/Components/LandingPage/index.css]: unnecessary whitespace

File: src/Components/LandingPage/index.css [hunk 3]
```diff
@@ -1,12 +1,70 @@
...
+    margin: 0 auto;
+}
+
+.main-container {
+    height: 60vh;
+    width: 90%;
+    min-height: 400px;
+    min-width: 691px;
+    padding: 5vh 0;
+    margin: 0 auto;
```
Comment (sudokufan) [inline src/Components/LandingPage/index.css]: there's a mix throughout of px, % and vh - this is probably the best example as it's all on one class. It's always worth sticking with one and keeping it as consistent as possible throughout, but I can see the use cases for each one and why you've used it

File: src/Components/Message/index.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+
+export class Message extends React.Component {
+    constructor(props) {
+        super(props)
+        this.state = {'msg': this.props.text}
+    }
+
+    componentWillReceiveProps(nextProps) {
+        this.setState({'msg':nextProps.text})
```
Comment (sudokufan) [inline src/Components/Message/index.js]:
```suggestion
        this.setState({'msg': nextProps.text})
```
being very picky, I think this has a space (as above on line 6)

File: src/Components/TextInput/index.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import React from "react";
+
+export class TextInput extends React.Component {
+    render() {
+        return (
+            <input onKeyUp={this.props.action} type="text" name={this.props.value} maxLength={this.props.max} minLength={this.props.min} placeholder={this.props.placeholderText} />
```
Comment (sudokufan) [inline src/Components/TextInput/index.js]:
```suggestion
            <input onKeyUp={this.props.action} type="text" name={this.props.value} maxLength={this.props.max} 
            minLength={this.props.min} placeholder={this.props.placeholderText} />
```
very long line - I'd break the line as shown to make it more readable

File: src/Components/VisitorSignInPage/index.css [hunk 6]
```diff
@@ -0,0 +1,90 @@
+.main-container {
+    padding: 0 0 0 0;
+}
+
+.sign-in-greeting-container {
+    padding: 10px 0 15px 0;
+
```
Comment (sudokufan) [inline src/Components/VisitorSignInPage/index.css]:
```suggestion
```
unnecessary line of whitespace

File: src/Components/VisitorSignInPage/index.css [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    margin: 0 5% 0 0;
+}
+
+.sign-in-input-container > input:nth-child(2) {
+    width: 47.5%;
+}
+
+.sign-in-input-container > input:nth-child(3) {
+    width: 100%;
+    margin: 15px 0 0 0;
```
Comment (sudokufan) [inline src/Components/VisitorSignInPage/index.css]: same as before about % alongside px - again I can see the argument for it, just want to bring it up to make sure there isn't a better solution

File: src/index.js [hunk 8]
```diff
@@ -3,21 +3,19 @@ import ReactDOM from 'react-dom';
 import {Route, Link, BrowserRouter as Router, Switch } from 'react-router-dom';
```
Comment (sudokufan) [inline src/index.js]:
```suggestion
```
unnecessary whitespace

File: src/index.js [hunk 9]
```diff
@@ -3,21 +3,19 @@ import ReactDOM from 'react-dom';
...
 
 import {LandingPage} from './Components/LandingPage';
+import {VisitorSignInPage} from './Components/VisitorSignInPage';
```
Comment (sudokufan) [inline src/index.js]:
```suggestion
```
unnecessary whitespace

File: src/index.js [hunk 10]
```diff
@@ -3,21 +3,19 @@ import ReactDOM from 'react-dom';
...
             <Router>
                 <div>
                     <Switch>
-                        <Route exact path="/" component={LandingPage}/>
-                        {/*<Route path="/visitor" component={VisitorSignInPage}/>*/}
-                        {/*<Route path="/landingPage" component={LandingPage}/>*/}
+                        <Route exact path="/" component={LandingPage} />
+                        <Route path="/visitors" component={VisitorSignInPage} />
+                        <Route path="/" component={LandingPage} />
                         {/*<Route component={NotFound}/>*/}
```
Comment (sudokufan) [inline src/index.js]: is this supposed to be commented out?

File: src/Components/LandingPage/index.css (hunk 3 above)
Comment (bentleybiscuit) [reply src/Components/LandingPage/index.css]: You're absolutely right. We're going to leave it like this at the moment.

File: src/Components/TextInput/index.js (hunk 5 above)
Comment (bentleybiscuit) [reply src/Components/TextInput/index.js]: Yep, good idea.

File: src/Components/VisitorSignInPage/index.css (hunk 7 above)
Comment (bentleybiscuit) [reply src/Components/VisitorSignInPage/index.css]: Chose to do it this way instead of complete refactoring of forename and surname.

File: src/index.js (hunk 10 above)
Comment (bentleybiscuit) [reply src/index.js]: Nice eagle-eyeing!
